### DISCUSSION TITLE
Homebrew tries to un-tar wrong dependency filenames (wrong random prefixes) during installs

### DISCUSSION THREAD

**Discussion by jasonlo87** [Homebrew/discussions#1482] [Getting started]
So I'm running Homebrew 3.1.7 on MacOs Catalina, and I'm seeing the following behavior...

If I do `brew install yarn`, the install fails with the following output:

```
==> Installing dependencies for yarn: icu4c, openssl@1.1, nghttp2 and node
==> Installing yarn dependency: icu4c
==> Pouring icu4c-69.1.catalina.bottle.tar.gz
tar: Error opening archive: Failed to open '/Users/xxxxx/Library/Caches/Homebrew/downloads/d0d75ee8daa0f562076c0e85e37e3a2cb2ab1ecbe390defe0a38519579f995f9--icu4c-69.1.catalina.bottle.tar.gz'
Error: Failure while executing; `tar --extract --no-same-owner --file /Users/xxxxx/Library/Caches/Homebrew/downloads/d0d75ee8daa0f562076c0e85e37e3a2cb2ab1ecbe390defe0a38519579f995f9--icu4c-69.1.catalina.bottle.tar.gz --directory /private/tmp/d20210518-58725-j14tis` exited with 1. Here's the output:
tar: Error opening archive: Failed to open '/Users/xxxxx/Library/Caches/Homebrew/downloads/d0d75ee8daa0f562076c0e85e37e3a2cb2ab1ecbe390defe0a38519579f995f9--icu4c-69.1.catalina.bottle.tar.gz'
```
And if i check that downloads directory, it DOES have a tar.gz for icu4c, but it's prefixed differently. In the above it's trying to un-tar **d0d75ee8daa0f562076c0e85e37e3a2cb2ab1ecbe390defe0a38519579f995f9**--icu4c, whereas the actual file downloaded is **_ae98a35ec0a62dc76f69a11ae87dd39d0c2bf6fef79f3b1a4ad8ebcebfddd341_**--icu4c, **so it's trying to un-tar a file that doesn't exist.** See below:

```
xxxxxx:downloads xxxxxx$ ls ~/Library/Caches/Homebrew/downloads
319330d53dab436f7e6bbe478539f29d41abb6be79788b06615f28006a82b30c--icu4c-69.1.bottle_manifest.json	b3267bd6e8fb63722958befde9283be69809068ffa8e6b7f8a61bc21ef8e0517--yarn--1.22.10.all.bottle.tar.gz
870999c7961ea239c1327483aaf361f87e71f16f0210e79b6d5bfb908d135b92--node--16.1.0.catalina.bottle.tar.gz	eb47612bc041cc28c0201bef4674302b4259f108eaa373dd4377869cff233932--nghttp2-1.43.0.catalina.bottle.tar.gz
a6c30763ba02dc89f8e17eaf6cebb33db4ae0c5693b9986be57d61be6227e014--node-16.1.0.bottle_manifest.json	fe9f3f05126e3c22858ffe89ae2c95869bf48558392becf5093812ec048e6a8f--openssl
ae98a35ec0a62dc76f69a11ae87dd39d0c2bf6fef79f3b1a4ad8ebcebfddd341--icu4c--69.1.catalina.bottle.tar.gz	fef38f3bec74bb426ce140d42de72df92e638f2f77da48c0e28cb75439bcfb03--yarn-1.22.10.bottle_manifest.json
```

I can sidestep the issue by doing `brew install icu4c` myself, but that's not how it should work. This issue is also not isolated to this yarn package, I have it with others as well (`brew install node` for example, also fails untarring icu4c and openssl@1.1).

Anybody have any ideas what the issue is?

**Comment by SMillerDev** [comment]
Please post the output of `brew config` and `brew doctor`

**Reply by jasonlo87** [reply]
Brew Doctor:
`$ brew doctor`
`Your system is ready to brew.`

Brew Config:
`$ brew config`
`HOMEBREW_VERSION: 3.1.8`
`ORIGIN: https://github.com/Homebrew/brew`
`HEAD: e0bfcdaf258dadaaf8c102d4cb51f23b6fdb2c4c`
`Last commit: 4 days ago`
`Core tap ORIGIN: https://github.com/Homebrew/homebrew-core`
`Core tap HEAD: cb5ca41204e516681de6046ae4fe9ccb4d77257f`
`Core tap last commit: 7 minutes ago`
`Core tap branch: master`
`HOMEBREW_PREFIX: /usr/local`
`HOMEBREW_BOTTLE_DOMAIN: <our-artifactory>`
`HOMEBREW_CASK_OPTS: []`
`HOMEBREW_MAKE_JOBS: 12`
`all_proxy: <our-proxy-address>`
`http_proxy: <our-proxy-address>`
`https_proxy: <our-proxy-address>`
`no_proxy: <our-domain-list>`
`Homebrew Ruby: 2.6.3 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/bin/ruby`
`CPU: dodeca-core 64-bit kabylake`
`Clang: 12.0.0 build 1200`
`Git: 2.24.3 => /Library/Developer/CommandLineTools/usr/bin/git`
`Curl: 7.64.1 => /usr/bin/curl`
`macOS: 10.15.7-x86_64`
`CLT: 12.0.0.32.29`
`Xcode: N/A`

**Reply by SMillerDev** [reply]
Seeing how you have a custom bottle domain and a proxy I can't really help you if those bottles don't match what Homebrew serves. You could check `https://formulae.brew.sh/api/formula/<name>.json` what bottles are the official Homebrew ones.
